### PR TITLE
[fix](row-policy) row policy supports external catalog

### LINKED ISSUE #19537: [Bug] Row policy can not apply to external catalog
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Version

master

### What's Wrong?

Row policy can not apply to external catalog

### What You Expected?

Support it

### How to Reproduce?

_No response_

### Anything Else?

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by stalary**
# Proposed changes

Issue Number: close https://github.com/apache/doris/issues/19537

## Problem summary

Row policy support external catalog

## Checklist(Required)

* [x] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [x] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by morningman** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/analysis/StmtRewriter.java [hunk 1]
```diff
@@ -1322,7 +1322,7 @@ public static boolean rewriteByPolicy(StatementBase statementBase, Analyzer anal
             if (dbName == null) {
                 dbName = analyzer.getDefaultDb();
             }
-            Database db = currentEnv.getInternalCatalog().getDbOrAnalysisException(dbName);
+            DatabaseIf db = currentEnv.getCurrentCatalog().getDbOrAnalysisException(dbName);
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/StmtRewriter.java:1325]: You need to use `tableRef.getName().getCtl()`

**Comment by stalary** [comment]
run buildall

**Comment by stalary** [comment]
run p0

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**